File: backend/src/routes/nbnhhsh.rs
use rocket::serde::json::Json;
use rocket::serde::{Deserialize, Serialize};
use utoipa::ToSchema;

/// Request body for nbnhhsh guess
#[derive(Debug, Deserialize, ToSchema)]
#[serde(crate = "rocket::serde")]
pub struct GuessRequest {
    /// 要猜测的文本
    pub text: String,
}

/// Response item from nbnhhsh API
#[derive(Debug, Serialize, Deserialize, Clone, ToSchema)]
#[serde(crate = "rocket::serde")]
pub struct GuessResult {
    /// 原始文本
    pub name: String,
    /// 翻译结果
    #[serde(skip_serializing_if = "Option::is_none")]
    pub trans: Option<Vec<String>>,
    /// 正在输入的候选
    #[serde(skip_serializing_if = "Option::is_none")]
    pub inputting: Option<Vec<String>>,
}

/// Proxy endpoint for nbnhhsh guess API
#[utoipa::path(
    post,
    path = "/api/nbnhhsh/guess",
    request_body = GuessRequest,
    responses(
        (status = 200, description = "成功获取猜测结果", body = Vec<GuessResult>)
    ),
    tag = "工具接口"
)]
#[post("/nbnhhsh/guess", data = "<request>")]
pub async fn guess(request: Json<GuessRequest>) -> Json<Vec<GuessResult>> {
    let client = reqwest::Client::new();
    
    let result = client
        .post("https://lab.magiconch.com/api/nbnhhsh/guess")
        .json(&serde_json::json!({ "text": request.text }))
        .send()
        .await;

    match result {
        Ok(response) => {
            if let Ok(data) = response.json::<Vec<GuessResult>>().await {
                Json(data)
            } else {
                Json(vec![])
            }
        }
        Err(_) => Json(vec![]),
    }
}

File: backend_ts/src/routes/nbnhhsh.ts
import { Elysia, t } from "elysia";
import { responsePlugin } from "@/plugins/response";

/**
 * nbnhhsh 工具路由
 *
 * 提供缩写翻译服务的代理接口
 * - POST /nbnhhsh/guess - 猜测缩写含义
 */
export const nbnhhshRoutes = new Elysia({ prefix: "/nbnhhsh" })
  .use(responsePlugin())

  // POST /nbnhhsh/guess - 猜测缩写
  .post(
    "/guess",
    async ({ body, success, error, set }) => {
      try {
        // 调用 nbnhhsh API
        const response = await fetch("https://lab.magiconch.com/api/nbnhhsh/guess", {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify({ text: body.text }),
        });

        if (!response.ok) {
          set.status = 500;
          return error(500, "Failed to fetch from nbnhhsh API");
        }

        const data = await response.json();
        return success(data);
      } catch (err) {
        console.error("Error calling nbnhhsh API:", err);
        set.status = 500;
        return error(500, "Failed to guess abbreviation");
      }
    },
    {
      body: t.Object({
        text: t.String(),
      }),
    },
  );
